Drivers/MotorChannel/Src/ZP_D_PWMChannel.cpp

```diff
@@ -0,0 +1,39 @@
+#include "ZP_D_PWMChannel.hpp"
+
+PWMChannel::PWMChannel(TIM_HandleTypeDef* timer,
+                        uint16_t timerChannel) :
+                        TIMER(timer),
+                        TIMER_CHANNEL(timerChannel)
+{
+	/* Sets up the PWM, pre-scaler, duty cycle ranges,
+	 * and starts the timer*/
+	PERIOD_TICKS = TIMER->Init.Period;
+	minSignal= PERIOD_TICKS * 0.05; // sets counts for 5% duty cycle
+	maxSignal = PERIOD_TICKS * 0.10; // sets counts for 10% duty cycle
+
+	// Calculate new pre-scaler
+	uint16_t prescaler = (CLOCK_FREQUENCY / DESIRED_FREQUENCY
+			/ PERIOD_TICKS) - 1;
+
+	__HAL_TIM_SET_PRESCALER(TIMER, prescaler);
+	HAL_TIM_PWM_Start(TIMER, TIMER_CHANNEL);
+
+
+	isSetup = true;
+}
```

Comment:
We can use the initialization list for a whole bunch of these.
```suggestion
PWMChannel::PWMChannel(TIM_HandleTypeDef* timer,
                        uint16_t timerChannel) :
                        MIN_SIGNAL(TIMER->Init.Period * 0.05),  // sets counts for 5% duty cycle
                        MAX_SIGNAL(TIMER->Init.Period * 0.10),  // sets counts for 10% duty cycle
                        TIMER(timer),
                        TIMER_CHANNEL(timerChannel)
{
	// Calculate new pre-scaler
	uint16_t prescaler = (CLOCK_FREQUENCY / DESIRED_FREQUENCY
			/ TIMER->Init.Period) - 1;

	__HAL_TIM_SET_PRESCALER(TIMER, prescaler);
	HAL_TIM_PWM_Start(TIMER, TIMER_CHANNEL);
}
```